### DISCUSSION TITLE
Hide command line popup for :nohl

### DISCUSSION THREAD

**Discussion by convers39** [folke/noice.nvim#609] [Q&A] [answered]
I have a mapping for no highlight as below, but it pops up the command line every time when pressing esc..  Is there any way to avoid popping up the command line except to disable it totally?

```lua
    ["<ESC>"] = { "<cmd>nohl<cr>", "No highlight", opts = {} },
```

https://github.com/folke/noice.nvim/assets/42597395/e2391458-0875-4142-a995-9b1d8ce9e7fd

**Comment by folke** [comment]
That's not normal. `<cmd>` never triggers the cmdline, so it's probably another keymap that is doing this because it has `:`.

You can try adding `silent = true`, but again, for this keymap it should already be silent.

**Reply by convers39** [reply, marked as answer]
@folke Thank you for your reply.

After some trial and error, it appears that I have mapping for both `<Esc>` and `<ESC>` accidentally.

I use nvchad and it has the default mapping for `<Esc>` as `:noh<cr>`, and I realized that `:` triggers the popup while `<cmd>` does not.

The solution is either one of 
- disable nvchad mapping for `<Esc>` (not `<ESC>`), then define my own either for `<Esc>` or `<ESC>`
- override mapping for `<Esc>` (not `<ESC>`, otherwise `<Esc>` will still be triggered, that is the reason in my case)

No the popup disappears!!

<img width="579" alt="image" src="https://github.com/folke/noice.nvim/assets/42597395/37648589-23c4-42ab-857a-1ab11841a71c">
